cmd/promtool/main.go

```diff
@@ -326,3 +349,101 @@ func CheckMetrics() int {
 
 	return 0
 }
+
+// QueryInstant performs an instant query against prometheus server.
+func QueryInstant(url *url.URL, query string) int {
+	config := api.Config{
+		Address: url.String(),
+	}
+
+	// Create new client.
+	c, err := api.NewClient(config)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error creating API client:", err)
+		return 1
+	}
+
+	// Run query against client.
+	api := v1.NewAPI(c)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	val, err := api.Query(ctx, query, time.Now())
+	cancel()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "query error:", err)
+		return 1
+	}
+
+	fmt.Println(val.String())
+
+	return 0
+}
+
+// QueryRange performs a range query against a prometheus server.
+func QueryRange(url *url.URL, query string, start string, end string) int {
+	config := api.Config{
+		Address: url.String(),
+	}
+
+	// Create new client.
+	c, err := api.NewClient(config)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "error creating API client:", err)
+		return 1
+	}
+
+	var stime, etime time.Time
+
+	if end == "" {
+		etime = time.Now()
+	} else {
+		etime, err = parseTime(end)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "error parsing end time:", err)
+			return 1
+		}
+	}
+
+	if start == "" {
+		stime = etime.Add(-5 * time.Minute)
+	} else {
+		stime, err = parseTime(start)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "error parsing start time:", err)
+		}
+	}
+
+	if !stime.Before(etime) {
+		fmt.Fprintln(os.Stderr, "start time is not before end time")
+	}
+
+	resolution := math.Max(math.Floor(etime.Sub(stime).Seconds()/250), 1)
+	// Convert seconds to nanoseconds such that time.Duration parses correctly.
+	step := time.Duration(resolution * math.Pow(10, 9))
```

Comment:
`math.Pow(10, 9)` can just be `1e9`.

Or better, do `time.Duration(resolution * time.Second)` because that keeps it independent of the integer representation of `time.Duration`.